src/key_store/mod.rs

```diff
@@ -0,0 +1,184 @@
+//! A storage solution for cryptographic key material.
+//!
+//! This module provides access to the `KeyStore` struct, which manages the
+//! storage of `CredentialBundle` and `KeyPackageBundle` instances. The
+//! development of this module is tracked in #337, which also includes a
+//! roadmap.
+//!
+//! The current key store enables the storage of `CredentialBundle` instances,
+//! and grants access to `CredentialBundle` references via the
+//! `SignaturePublicKey` of the corresponding `Credential`.
+//!
+//! A `KeyStore` is meant to be used across multiple `ManagedGroup` instances to
+//! allow sharing the same `CredentialBundle`. If this is not desired, multiple
+//! `KeyStore` instances can be used across groups.
+//!
+//! # Example
+//!
+//! A simple example for the generation and the retrieval of a
+//! `CredentialBundle`.
+//!
+//! ```
+//! use openmls::prelude::*;
+//!
+//! let key_store = KeyStore::default();
+//!
+//! // Generate a credential bundle.
+//! let alice_credential = key_store
+//!     .generate_credential(
+//!         "Alice".into(),
+//!         CredentialType::Basic,
+//!         SignatureScheme::ED25519,
```

Comment:
Would be nice if we could use the `ciphersuite_name` here as well.

Reply:
We can and I changed it so that we do.